### DISCUSSION TITLE
Understanding Delay in Inference

### DISCUSSION THREAD

**Discussion by BKTidswell** [talmolab/sleap#1677] [Help!]
Hello, I'm just trying to understand why there's a long delay in inference, and what to do to speed it up. When I run inference it takes about 8 minutes to process a single 600 frame video with my current bottom up model. What's confusing me is that I see a message like this:
`{"n_processed": 601, "n_total": 601, "elapsed": 31.74941849708557, "rate": 18.92948055269638, "eta": 0.0}`

And then the process continues for around another 500 seconds without showing any output, with SLEAP barely using an GPU, mostly just CPU and RAM. 

I do also see:
`System:
GPUs: 1/1 available
  Device: /physical_device:GPU:0
         Available: True
        Initalized: False
     Memory growth: True`
     
 Does it need to initialize, is that what the issue is? What is SLEAP doing after I see the last message shown above?

**Comment by roomrys** [comment]
Hi @BKTidswell,

The [most in-depth discussion we have on this](https://github.com/talmolab/sleap/discussions/833) ended with:
> To simplify the info in the discussion here, two things really helped us achieve speed ups from 5-10 FPS to over 400FPS:
> 
> 1. Correct drivers: SLEAP/other GPU dependent software doesn't seem to crash as much as struggle if the non-optimal drivers are being referenced in your linux terminal.
> 2. Separating out CPU and GPU workloads as @talmo suggested. This helps enormously with people's workflows. On A40s it appears that doing both CPU/GPU tasks in the same sleap-track call is sufficiently fast, but each step is about 2x as fast if they are separated out.

Usually we recommend updating the GPU drivers which has worked most of the time.

Liezl
